### PR TITLE
Basic Flask Project Structure, Simple Faculty API with ElasticSearch

### PR DESCRIPTION
**Pull request by GarryBullock**
Hey all,

This is a proposed project structure, as well as a basic api for faculty members. Essentially the start of the data ingestion so we can feed the data dumps into our system.

To run this, you will need to set up a few things:
 - [ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html) which also involves installing JDK 1.8 (OpenJDK works too).
 - [MiniConda](https://conda.io/docs/glossary.html#miniconda-glossary): A virtual environment and package manager for python. Essentially so we can make sure each of us is using the same version of python and modules. I've provided an environment.yml that you can use to build a duplicate environment. Instructions on that [here](https://conda.io/docs/user-guide/tasks/manage-environments.html). If you'd rather use virtualenv that's cool too but you'll need to manage installing everything required yourself.

This was mostly just me playing around with flask and seeing if elasticsearch was going to be okay for us. If you have questions let me know.

### CODE REVIEW

**Comment by GarryBullock** [comment]
Added a new update.

Finished the (basic) work on the Faculty API prototype. Allows you to call /faculty using a post request with JSON and process the entire faculty file that they gave us. Takes about ~2.5 seconds which seems fine. I can show you guys next time we meet.

screenshot from 2018-02-05 22-55-18

I'd like everyone to actually make a comment here so I know you've reviewed it. Also, if someone wants to do something similar for the grants and publications, that'd be great.

File: bfex/components/data_ingestor.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        :exception TypeError: If the json_data is not a sequence object. The expected type is a List.
+        """
+        if not isinstance(json_data, Sequence):
+            raise TypeError("Expected a Sequence, but got a {}", type(json_data))
+
+        count = 0
+        for faculty_member in json_data:
+            count += 1
+            DataIngester.create_faculty(faculty_member, write)
+        print("Ingested {} faculty members".format(count))
```
Comment (GarryBullock) [inline bfex/components/data_ingestor.py]: This was just for informational purposes. We could log this info if we wanted.

**Comment by 363879721** [comment]
looks good to me too!

File: bfex/models.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    Publication.init()
+
+
+if __name__ == "__main__":
+    from elasticsearch_dsl.connections import connections
+    connections.create_connection(hosts=["localhost"])
+
+    Faculty.init()
+
+    prof = Faculty(meta={'id': 42}, name="Garry Bullock", email="gbullock@ualberta.ca", faculty_id=42)
```
Comment (MandyMeindersma) [inline bfex/models.py]: :P

File: bfex/templates/faculty.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    width: 100%;
+}
+
+td, th {
+    border: 1px solid #dddddd;
+    text-align: left;
+    padding: 8px;
+}
+
+tr:nth-child(even) {
```
Comment (MandyMeindersma) [inline bfex/templates/faculty.html]: td and th are? table design? table header?

File: bfex/templates/faculty.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    background-color: #dddddd;
+}
+</style>
+</head>
+<body>
+
+<table>
+  <tr>
```
Comment (MandyMeindersma) [inline bfex/templates/faculty.html]: table row?

**Comment by MandyMeindersma** [comment]
Kibana is neat. I have never heard of it before. 

It looks great.You have to remind me what a couple of your acronyms though.

I'll work on setting things up this weekend.

Thank you :pray:

File: environment.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
+name: BFEX
+channels:
```
Comment (MandyMeindersma) [inline environment.yml]: gracias for this! MiniConda is beautiful

File: bfex/templates/faculty.html (hunk 3 above)
Comment (GarryBullock) [reply bfex/templates/faculty.html]: Td is a standard html table cell.

File: bfex/templates/faculty.html (hunk 4 above)
Comment (GarryBullock) [reply bfex/templates/faculty.html]: Yes

**Pull request merged**